protocol/chainlib/grpc.go

```diff
@@ -200,11 +200,12 @@ func (apil *GrpcChainListener) Serve(ctx context.Context) {
 		var relayReply *pairingtypes.RelayReply
 		metricsData := metrics.NewRelayAnalytics("NoDappID", apil.endpoint.ChainID, apiInterface)
 		if relayReply, _, err = apil.relaySender.SendRelay(ctx, method, string(reqBody), "", "NoDappID", metricsData); err != nil {
-			go apil.logger.AddMetric(metricsData, err != nil)
+			go apil.logger.AddMetricForGrpc(metricsData, false, ctx)
 			errMasking := apil.logger.GetUniqueGuidResponseForError(err, msgSeed)
 			apil.logger.LogRequestAndResponse("http in/out", true, method, string(reqBody), "", errMasking, msgSeed, err)
 			return nil, utils.LavaFormatError("Failed to SendRelay", fmt.Errorf(errMasking), nil)
 		}
+		go apil.logger.AddMetricForGrpc(metricsData, true, ctx)
```

Comment:
also, one thing to note here is that the context is canceled when we return the response (by the user) as they run in a routine, if you need a context. you would need to create a new background context inside the AddMetric.

Reply:
I saw you need this only for grpc for the metadata. as the context is canceled by the user you will need to extract the meta data before you start the routine, we are going to use the meta data later on anyway so you can just unload the entire meta data object to when we get the request and use the meta data in your go routine.